lib/identity/hostdata/config_builder.rb

```diff
@@ -63,12 +64,42 @@ def initialize
         @key_types = {}
       end
 
-      def add(key, type: :string, allow_nil: false, enum: nil, options: {})
-        value = @read_env[key]
+      # @param key [Symbol] secret property name
+      # @param secrets_manager_name [String] if present, the secret_id for Secrets Manager to get
+      #   the value from in a deployed environment
+      # @param secrets_manager_local_name [String] if present, the key to look up a local equivalent
+      #   of a Secrets Manager secret in local development. When absent, +secrets_manager_name+ is
+      #   used locally
+      # @param type [Symbol] secret type, used to parse raw value
+      # @param allow_nil [Boolean] whether or not a nil value is allowed
+      # @param enum [nil, Array] list of allowed values
+      # @param options [Hash] options hash, passed to per-type converter
+      def add(
+        key,
+        secrets_manager_name: nil,
+        secrets_manager_local_name: nil,
+        type: :string,
+        allow_nil: false,
+        enum: nil,
+        options: {}
+      )
+        value = if secrets_manager_name
+          if Identity::Hostdata.in_datacenter?
```

Comment:
I think we may want to have an `ENV` option to opt in/out too since there could be cases where we want to test locally?